head/src/main/kotlin/io/qalipsis/core/head/report/thymeleaf/MediaReplacedElementFactory.kt

```diff
@@ -72,4 +73,42 @@ internal class MediaReplacedElementFactory(private val superFactory: ReplacedEle
 
     override fun setFormSubmissionListener(listener: FormSubmissionListener?) =
         superFactory.setFormSubmissionListener(listener)
+
+    /**
+     * Converts the svg image to a convenient png byte array format.
+     *
+     * @param imageSrc the image string content
+     */
+    private fun svgToPng(imageSrc: String): ByteArray {
+        val base64ByteArray = Base64.getDecoder().decode(imageSrc)
+        // Set the input source as the SVG data.
+        val input = TranscoderInput(ByteArrayInputStream(base64ByteArray))
+        // Create an output stream for the converted image.
+        return ByteArrayOutputStream().use { outputStream ->
+            val output = TranscoderOutput(outputStream)
+            // Perform the SVG to PNG conversion.
+            val transcoder = PNGTranscoder()
+            transcoder.transcode(input, output)
+            // Get the PNG image as a byte array.
+            outputStream.toByteArray()
+        }
+    }
+
+    private companion object {
+
+        val logger = logger()
+
+        const val DEFAULT_SCALE_WIDTH = 285
```

Comment:
Please document the constants and why those values are not arbitrary.

Reply:
These values here are specifically for the icons. I think I set a fixed width and height for the chart images. I don't think i should give users the choice to pick the image size they want I guess

Reply:
That's not what I mean, but a developer needs to know why those values are the right ones. Also document for what they are used (icons, as you indicated).